actors/migration/nv14/test/deal_label_migration_test.go

```diff
@@ -0,0 +1,273 @@
+
+}
+
+func publishDealv5(t *testing.T, v *vm5.VM, provider, dealClient, minerID addr.Address, dealLabel string,
+	pieceSize abi.PaddedPieceSize, verifiedDeal bool, dealStart abi.ChainEpoch, dealLifetime abi.ChainEpoch,
+) *market5.PublishStorageDealsReturn {
+	deal := market5.DealProposal{
+		PieceCID:             tutil.MakeCID(dealLabel, &market.PieceCIDPrefix),
+		PieceSize:            pieceSize,
+		VerifiedDeal:         verifiedDeal,
+		Client:               dealClient,
+		Provider:             minerID,
+		Label:                dealLabel,
+		StartEpoch:           dealStart,
+		EndEpoch:             dealStart + dealLifetime,
+		StoragePricePerEpoch: abi.NewTokenAmount(1 << 20),
+		ProviderCollateral:   big.Mul(big.NewInt(2), vm.FIL),
+		ClientCollateral:     big.Mul(big.NewInt(1), vm.FIL),
+	}
+
+	publishDealParams := market5.PublishStorageDealsParams{
+		Deals: []market5.ClientDealProposal{{
+			Proposal: deal,
+			ClientSignature: crypto.Signature{
+				Type: crypto.SigTypeBLS,
+			},
+		}},
+	}
+	result := vm5.RequireApplyMessage(t, v, provider, builtin.StorageMarketActorAddr, big.Zero(), builtin.MethodsMarket.PublishStorageDeals, &publishDealParams, t.Name())
+	require.Equal(t, exitcode.Ok, result.Code)
+
+	expectedPublishSubinvocations := []vm5.ExpectInvocation{
+		{To: minerID, Method: builtin.MethodsMiner.ControlAddresses, SubInvocations: []vm5.ExpectInvocation{}},
+		{To: builtin.RewardActorAddr, Method: builtin.MethodsReward.ThisEpochReward, SubInvocations: []vm5.ExpectInvocation{}},
+		{To: builtin.StoragePowerActorAddr, Method: builtin.MethodsPower.CurrentTotalPower, SubInvocations: []vm5.ExpectInvocation{}},
+	}
+
+	if verifiedDeal {
+		expectedPublishSubinvocations = append(expectedPublishSubinvocations, vm5.ExpectInvocation{
+			To:             builtin.VerifiedRegistryActorAddr,
+			Method:         builtin.MethodsVerifiedRegistry.UseBytes,
+			SubInvocations: []vm5.ExpectInvocation{},
+		})
+	}
+
+	vm5.ExpectInvocation{
+		To:             builtin.StorageMarketActorAddr,
+		Method:         builtin.MethodsMarket.PublishStorageDeals,
+		SubInvocations: expectedPublishSubinvocations,
+	}.Matches(t, v.LastInvocation())
+
+	return result.Ret.(*market.PublishStorageDealsReturn)
+}
+
+func checkMarket5ProposalsEtcState(t *testing.T, proposals *adt.Array, states *adt.Array, pendingProposals *adt.Set,
+	dealID abi.DealID, inProposals bool, inStates bool, inPendingProposals bool) {
+	var dealprop5 market5.DealProposal
+	found, err := proposals.Get(uint64(dealID), &dealprop5)
+	require.NoError(t, err)
+	require.Equal(t, found, inProposals)
+	found, err = states.Get(uint64(dealID), nil)
+	require.NoError(t, err)
+	require.Equal(t, found, inStates)
+	dealpropcid, err := dealprop5.Cid()
+	require.NoError(t, err)
+	found, err = pendingProposals.Has(abi.CidKey(dealpropcid))
+	require.NoError(t, err)
+	require.Equal(t, found, inPendingProposals)
+}
+
+// there MUST be a way to consolidate these...
```

Comment:
You could create a function parameter that gets a proposal and returns the cid, returning cid.Undef if not found.  You could then get found info from this and the pending proposals key.

That said this is fine as is.  Either way remove this comment before merging.